optuna/study/study.py

```diff
@@ -569,6 +633,9 @@ def df(x):
                 :class:`~optuna.trial.TrialState.PRUNED`.
                 ``trial`` is a trial number but no
                 trial exists with that number.
+
+        Returns:
+            A :class:`~optuna.trial.FrozenTrial` representing the resulting trial.
```

Comment:
Should we guarantee that this is a (deep) copy, and that the caller is free to modify it? It's probably important for the internal usage of `Study.tell` as well.

Reply:
Thanks!